src/pymor/models/transforms.py

```diff
@@ -0,0 +1,107 @@
+import numpy as np
+from scipy.linalg import inv, null_space
+
+from pymor.models.interface import Model
+
+
+class MoebiusTransform(Model):
+    """A Moebius transform operator.
+
+    Maps complex square matrices to complex square matrices.
+    The size is given by dim. Scalars are 1x1. The transform coefficients can be normalized.
+    Contains a method for constructing an inverse mapping, as well as a constructor that takes three
+    points and their images. The Moebius transforms form a group under composition, so __matmul__ is
+    defined to yield Moebius Transforms, if both factors are Moebius Transforms.
+    """
+
+    def __init__(self, coefficients, dim=1, normalize=False, source_id=None, range_id=None, name=None):
+        assert len(coefficients) == 4
+        coefficients = np.array(coefficients, dtype=complex)
+        assert coefficients[0]*coefficients[3] != coefficients[1]*coefficients[2]
+
+        if normalize:
+            coefficients /= np.sqrt(coefficients[0]*coefficients[3]-coefficients[2]*coefficients[3])
+
+        self.__auto_init(locals())
+
+    @classmethod
+    def from_points(cls, w, z=[0, 1, np.inf], dim=1, source_id=None, range_id=None, name=None):
+        assert len(z) == 3
+        assert len(w) == 3
+
+        A = np.zeros([3, 4])
+        for i in range(3):
+            if np.isinf(w[i]) and np.isinf(z[i]):
+                A[i] = [0, 0, -1, 0]
+            elif np.isinf(w[i]):
+                A[i] = [0, 0, -z[i], -1]
+            elif np.isinf(z[i]):
+                A[i] = [1, 0, -w[i], 0]
+            else:
+                A[i] = [z[i], 1, -w[i]*z[i], -w[i]]
+
+        return cls(null_space(A).ravel(), dim=dim, source_id=source_id, range_id=range_id, name=name)
+
+    def inverse(self, normalize=False):
+        a, b, c, d = self.coefficients
+        coefficients = np.array([d, -b, -c, a])
+        return MoebiusTransform(
+            coefficients,
+            dim=self.dim,
+            normalize=normalize,
+            source_id=self.range_id,
+            range_id=self.source_id,
+            name=self.name + '_inverse',
+        )
+
+    def __call__(self, x):
+        a, b, c, d = self.coefficients
+        x = np.array(x, dtype=complex)
+        x = np.atleast_2d(x)
+        assert x.shape[0] == x.shape[1]
+        I = np.eye(x.shape[0])
+        if c != 0 and np.allclose(x, -d * I / c):
+            return np.inf * I
+        elif c != 0 and np.isinf(x).any():
+            return a * I / c
+        elif c == 0 and np.isinf(x).any():
+            return np.inf * I
+        else:
+            return (a * x + b * I) @ inv(c * x + d * I)
```

Comment:
It would be good to use `solve` instead of `inv`.

Reply:
I went back to supporting just scalars, because I realised that a thorough implementation of matrix maps would need more and there is no use case at the moment. I went ahead to allow one-dimensional arrays or lists as function inputs so that frequency vectors can be mapped.